File: app/src/main/java/com/coinninja/coinkeeper/service/BroadcastTransactionService.java
package com.coinninja.coinkeeper.service;

import android.app.IntentService;
import android.content.Intent;

import com.coinninja.coinkeeper.model.dto.CompletedBroadcastDTO;
import com.coinninja.coinkeeper.service.runner.SaveTransactionRunner;
import com.coinninja.coinkeeper.util.DropbitIntents;

import javax.inject.Inject;

import androidx.annotation.Nullable;
import dagger.android.AndroidInjection;

public class BroadcastTransactionService extends IntentService {
    public static final String TAG = BroadcastTransactionService.class.getName();

    @Inject
    SaveTransactionRunner runner;

    public BroadcastTransactionService() {
        this(TAG);
    }

    public BroadcastTransactionService(String name) {
        super(name);
    }

    @Override
    public void onCreate() {
        AndroidInjection.inject(this);
        super.onCreate();
    }

    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        if (intent.hasExtra(DropbitIntents.EXTRA_COMPLETED_BROADCAST_DTO)) {
            CompletedBroadcastDTO completedBroadcastActivityDTO = intent.getParcelableExtra(DropbitIntents.EXTRA_COMPLETED_BROADCAST_DTO);
            runner.setCompletedBroadcastActivityDTO(completedBroadcastActivityDTO);
            runner.run();
        }
    }

    public void setRunner(SaveTransactionRunner runner) {
        this.runner = runner;
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/service/BroadcastTransactionService.kt
package com.coinninja.coinkeeper.service

import android.app.IntentService
import android.content.Intent
import com.coinninja.coinkeeper.model.dto.CompletedBroadcastDTO
import com.coinninja.coinkeeper.service.runner.SaveTransactionRunner
import com.coinninja.coinkeeper.util.DropbitIntents
import dagger.android.AndroidInjection
import javax.inject.Inject

class BroadcastTransactionService @JvmOverloads constructor(name: String = TAG) : IntentService(name) {

    @Inject
    internal lateinit var runner: SaveTransactionRunner

    override fun onCreate() {
        AndroidInjection.inject(this)
        super.onCreate()
    }

    public override fun onHandleIntent(intent: Intent?) {
        intent?.let { intent ->
            if (intent.hasExtra(DropbitIntents.EXTRA_COMPLETED_BROADCAST_DTO)) {
                val completedBroadcastActivityDTO = intent.getParcelableExtra<CompletedBroadcastDTO>(DropbitIntents.EXTRA_COMPLETED_BROADCAST_DTO)
                runner.setCompletedBroadcastActivityDTO(completedBroadcastActivityDTO)
                runner.run()
            }
        }
    }

    companion object {
        val TAG = BroadcastTransactionService::class.java.name
    }
}
